actors/market/src/lib.rs

```diff
@@ -234,16 +235,34 @@ impl Actor {
             ));
         }
 
+        struct CheckedDeal {
+            proposal: DealProposal,
+            check_passed: bool,
+        }
+        // Deals that passed `AuthenticateMessage` and other state-less checks.
+        let mut checked_deals: Vec<CheckedDeal> = Vec::with_capacity(params.deals.len());
+
         let baseline_power = request_current_baseline_power(rt)?;
         let (network_raw_power, _) = request_current_network_power(rt)?;
 
+        for (di, deal) in params.deals.into_iter().enumerate() {
+            let mut check_passed = true;
+            if let Err(e) = validate_deal(rt, &deal, &network_raw_power, &baseline_power) {
```

Comment:
Nit: since we pass `rt` here it's not obvious that validation can't load state or have side effects. Consider passing the policy and other values explicitly.

Reply:
Well it needs to perform a `send`, so I'm not sure how achievable that is. We could rename to something like `validate_deal_stateless` for clarity & some weak future-proofing?